jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/BaseImageParameters.java

```diff
@@ -25,26 +25,23 @@
 import org.gradle.api.Action;
 import org.gradle.api.model.ObjectFactory;
 import org.gradle.api.provider.ListProperty;
-import org.gradle.api.provider.Property;
-import org.gradle.api.provider.Provider;
 import org.gradle.api.tasks.Input;
 import org.gradle.api.tasks.Nested;
 import org.gradle.api.tasks.Optional;
 
 /** Object in {@link JibExtension} that configures the base image. */
-public class BaseImageParameters {
+public class BaseImageParameters extends ImageParameters {
```

Comment:
We are not really a fan of inheritance for this kind of usage. I'd usually add `ImageParameters` as a field. But then, the parameter structure doesn't match the actual Jib configuration. (That is, `AuthParameters` should be a field as we have `image.auth`.) I think it's still fine to add the cred helper parameters as a field. What do you think?

Reply:
Hmm I think I get what you mean. I was trying to reduce the code duplication between `BaseImageParameters` and `TargetImageParameters`. I guess making `ImageParameters` a field would mean Jib configuration might have to change? e.g to configure auth we might need to do something like `from.image.auth` as opposed to `from.auth`?

If that's the case, I'm happy to revert and keep as it was before while making `CredHelperParameters` fields in both
`BaseImageParameters` and `TargetImageParameters`.

Reply:
> I guess making `ImageParameters` a field would mean Jib configuration might have to change? e.g to configure auth we might need to do something like `from.image.auth` as opposed to `from.auth`?

I think technically it is not impossible to have `ImageParameters` as a field while retaining the current Jib configuration structure. However, I don't really like doing that. I'd rather have the code structure in sync with Jib configuration.

So yeah, let's keep `CredHelperParameters` as a field.